### PR TITLE
LPD-28124 Adds an Upgrade process to delete orphan AssetListEntryAssetEntryRel records

### PR DESCRIPTION
**Pull request by lfbesada**
https://liferay.atlassian.net/browse/LPD-28124

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by lfbesada** [comment]
ci:test:upgrade

**Comment by lfbesada** [comment]
ci:report:13875306

**Comment by lfbesada** [comment]
ci:test:upgrade

**Comment by lfbesada** [comment]
ci:report:14724693

**Comment by lfbesada** [comment]
ci:test:upgrade

**Comment by lfbesada** [comment]
ci:report:14882422

**Comment by lfbesada** [comment]
ci:reevaluate:1257401_5440

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/upgrade/v2_1_1/test/AssetListEntryAssetEntryRelUpgradeProcessTest.java [hunk 1]
```diff
@@ -127,6 +129,53 @@ public void testUpgradeUpdatingPositionWithMultipleOrphanAssetEntryId()
...
+		SegmentsEntry segmentsEntry = SegmentsTestUtil.addSegmentsEntry(
+			_group.getGroupId());
+
+		AssetListTestUtil.addAssetListEntrySegmentsEntryRel(
+			_group.getGroupId(), _assetListEntry,
+			segmentsEntry.getSegmentsEntryId());
+
+		_addAssetListEntryAssetEntryRel(
+			segmentsEntry.getSegmentsEntryId(), journalArticle4);
+
```
Comment (ealonso) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/upgrade/v2_1_1/test/AssetListEntryAssetEntryRelUpgradeProcessTest.java:154]: @lfbesada I think the empty line is not needed if we call the same method, like you did for _addJournalArticle

Comment (lfbesada) [reply modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/upgrade/v2_1_1/test/AssetListEntryAssetEntryRelUpgradeProcessTest.java]: @ealonso Thanks for your feedback :heart: I have added it and rebased this branch with the current master. Could you please take a look? Thanks in advance!

**Comment by lfbesada** [comment]
ci:test:sf

**Comment by ealonso** [comment]
ci:forward

**Comment by lfbesada** [comment]
ci:report:17669243

**Comment by lfbesada** [comment]
ci:report:17669243

**Comment by lfbesada** [comment]
ci:forward:force

**Comment by lfbesada** [comment]
ci:report:17725577

**Pull request closed without merging**